### PR TITLE
Improve logging around report chunking

### PR DESCRIPTION
**Pull request by yunhanw-google**
-- Add detailed logging and correct level for multiple scenarios when encountering out of space during encoding report.

### CODE REVIEW

File: src/app/reporting/Engine.cpp [hunk 1]
```diff
@@ -204,16 +204,16 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
...
-                             ChipLogValueMEI(pathForRetrieval.mClusterId), err.Format());
-
                 // If error is not an "out of writer space" error, rollback and encode status.
                 // Otherwise, if partial data allowed, save the encode state.
                 // Otherwise roll back. If we have already encoded some chunks, we are done; otherwise encode status.
 
                 if (encodeState.AllowPartialData() && IsOutOfWriterSpaceError(err))
                 {
+                    ChipLogError(DataManagement,
+                        "Encoding is out of space, and partial data allowed, save the encode state for report chunking on clusterId: " ChipLogFormatMEI,
```
Comment (andy31415) [inline src/app/reporting/Engine.cpp:214]: is this an actuall `Error` ? why do we add `save the encode state ...` into the error text?

File: src/app/reporting/Engine.cpp [hunk 2]
```diff
@@ -204,16 +204,16 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
...
-                             "Error retrieving data from clusterId: " ChipLogFormatMEI ", err = %" CHIP_ERROR_FORMAT,
-                             ChipLogValueMEI(pathForRetrieval.mClusterId), err.Format());
-
                 // If error is not an "out of writer space" error, rollback and encode status.
                 // Otherwise, if partial data allowed, save the encode state.
                 // Otherwise roll back. If we have already encoded some chunks, we are done; otherwise encode status.
 
                 if (encodeState.AllowPartialData() && IsOutOfWriterSpaceError(err))
                 {
+                    ChipLogError(DataManagement,
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:213]: This should not be a ChipLogError.  Progress, or even detail...

File: src/app/reporting/Engine.cpp [hunk 3]
```diff
@@ -204,16 +204,16 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
...
+                        "Encoding is out of space, and partial data allowed, save the encode state for report chunking on clusterId: " ChipLogFormatMEI,
+                        ChipLogValueMEI(pathForRetrieval.mClusterId));
+                    ChipLogProgress(DataManagement, "Encoding is Out of space, and partial data allowed, save the encode state");
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:216]: Why do we have two logs here?

File: src/app/reporting/Engine.cpp [hunk 4]
```diff
@@ -230,6 +230,9 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
 
                     if (!IsOutOfWriterSpaceError(err))
                     {
+                        ChipLogError(DataManagement,
+                            "Fail to retrieve data, roll back and encode status on clusterId: " ChipLogFormatMEI ", err = %" CHIP_ERROR_FORMAT,
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:234]: Shouldn't this include the attribute id?  Could be useful....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/reporting/Engine.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/reporting/Engine.cpp]: not error, we need mention it is processing some metadata durinng report chunking.

File: src/app/reporting/Engine.cpp (hunk 2 above)
Comment (yunhanw-google) [reply src/app/reporting/Engine.cpp]: yes, updated with detail

File: src/app/reporting/Engine.cpp (hunk 3 above)
Comment (yunhanw-google) [reply src/app/reporting/Engine.cpp]: removed. thanks

File: src/app/reporting/Engine.cpp [hunk 5]
```diff
@@ -204,16 +204,16 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
...
-
                 // If error is not an "out of writer space" error, rollback and encode status.
                 // Otherwise, if partial data allowed, save the encode state.
                 // Otherwise roll back. If we have already encoded some chunks, we are done; otherwise encode status.
 
                 if (encodeState.AllowPartialData() && IsOutOfWriterSpaceError(err))
                 {
+                    ChipLogDetail(DataManagement,
+                                  "Encoding is out of space, and partial data allowed, save the encode state for report chunking "
+                                  "on clusterId: " ChipLogFormatMEI " ,attributeId: " ChipLogFormatMEI,
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:215]:
```suggestion
                                  "on clusterId: " ChipLogFormatMEI ", attributeId: " ChipLogFormatMEI,
```

File: src/app/reporting/Engine.cpp [hunk 6]
```diff
@@ -230,6 +230,11 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
 
                     if (!IsOutOfWriterSpaceError(err))
                     {
+                        ChipLogError(DataManagement,
+                                     "Fail to retrieve data, roll back and encode status on clusterId: " ChipLogFormatMEI
+                                     " ,attributeId: " ChipLogFormatMEI,
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:235]:
```suggestion
                                     ", attributeId: " ChipLogFormatMEI,
```

File: src/app/reporting/Engine.cpp [hunk 7]
```diff
@@ -204,16 +204,16 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
...
-                             ChipLogValueMEI(pathForRetrieval.mClusterId), err.Format());
-
                 // If error is not an "out of writer space" error, rollback and encode status.
                 // Otherwise, if partial data allowed, save the encode state.
                 // Otherwise roll back. If we have already encoded some chunks, we are done; otherwise encode status.
 
                 if (encodeState.AllowPartialData() && IsOutOfWriterSpaceError(err))
                 {
+                    ChipLogDetail(DataManagement,
+                                  "Encoding is out of space, and partial data allowed, save the encode state for report chunking "
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:214]: Might be clearer as:
```suggestion
                                  "List does not fit in packet, chunk between list items for clusterId: " ....
```

File: src/app/reporting/Engine.cpp [hunk 8]
```diff
@@ -239,6 +244,14 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
                             attributeReportIBs.Rollback(attributeBackup);
                         }
                     }
+                    else
+                    {
+                        ChipLogDetail(DataManagement,
+                                      "Encoding is out of space, roll back on clusterId: " ChipLogFormatMEI
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:250]:
```suggestion
                                      "Next attribute value does not fit in packet, roll back on clusterId: " ChipLogFormatMEI
```

File: src/app/reporting/Engine.cpp [hunk 9]
```diff
@@ -239,6 +244,14 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
...
+                                      " , attributeId: " ChipLogFormatMEI ", err = %" CHIP_ERROR_FORMAT,
```
Comment (bzbarsky-apple) [inline src/app/reporting/Engine.cpp:251]:
```suggestion
                                      ", attributeId: " ChipLogFormatMEI ", err = %" CHIP_ERROR_FORMAT,
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by chulspro — APPROVED** [review]

**Pull request merged**